jupyter_server_proxy/__init__.py

```diff
@@ -54,21 +59,29 @@ def _load_jupyter_server_extension(nbapp):
         serverproxy_config,
     )
 
-    icons = {}
+    icon_handlers = []
     for sp in server_processes:
         if sp.launcher_entry.enabled and sp.launcher_entry.icon_path:
-            icons[sp.name] = sp.launcher_entry.icon_path
+            icon_handlers.append(
+                (
+                    ujoin(base_url, f"server-proxy/icon/{sp.name}"),
+                    IconHandler,
+                    {"path": sp.launcher_entry.icon_path},
+                )
+            )
 
     nbapp.web_app.add_handlers(
         ".*",
         [
             (
-                ujoin(base_url, "server-proxy/servers-info"),
+                ujoin(base_url, "api/server-proxy/servers-info"),
                 ServersInfoHandler,
                 {"server_processes": server_processes},
             ),
-            (ujoin(base_url, "server-proxy/icon/(.*)"), IconHandler, {"icons": icons}),
-        ],
+            (ujoin(base_url, r"api/server-proxy"), ListServersAPIHandler),
+            (ujoin(base_url, r"api/server-proxy/(?P<name>.*)"), ServersAPIHandler),
+        ]
+        + icon_handlers,
```

Comment:
Welp, declaring (and testing against) an OpenAPI spec would be a good place to start, so that one doesn't have to _read the code_ to figure out what's going on. Prior art:
- https://github.com/jupyter-server/jupyter_server/blob/main/jupyter_server/services/api/api.yaml
- https://github.com/jupyterlab/jupyterlab_server/blob/main/jupyterlab_server/rest-api.yml

This can generate nice static docs, as well as help cleaning up some of the design decisions: if you can't express a pattern as a URL param or in a schema, maybe it's not a great idea.

It also then becomes possible to re-use that schema declaration to generate lightweight python and TS types, even if one doesn't do fully generated clients.

Reply:
Agree with OpenAPI suggestion. That would make it more cleaner. Actually, `servers-info` endpoint is only handling ["launcher" specific metadata](https://github.com/jupyterhub/jupyter-server-proxy/blob/407a1a4809c598dd30647a000343a2f32a145f1e/jupyter_server_proxy/api.py#L19-L30) whereas the new endpoints are handling [server specific metadata](https://github.com/jupyterhub/jupyter-server-proxy/blob/bac7ccce3bf84c529eeb78a72fcd85bbd8892490/jupyter_server_proxy/api.py#L76). So, I think it is logical to keep them as separate. However, I think we need to change the name of `servers-info` endpoint as it is bit misleading? `ServersAPIHandler` and `ListServersAPIHandler` can be merged into one using query parameters.

Reply:
@bollwyvl Cheers for the refs